From b1606a66972434ef9c80b54a33434cd50e4cf689 Mon Sep 17 00:00:00 2001
From: Christoph Reiter <reiter.christoph@gmail.com>
Date: Sun, 8 Aug 2021 10:17:35 +0200
Subject: [PATCH 110/N] expanduser: normpath paths coming from env vars

This makes sure we get the same paths as with related functions in pathlib.

Signed-off-by: Naveen M K <naveen521kk@gmail.com>
---
 Lib/ntpath.py | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/Lib/ntpath.py b/Lib/ntpath.py
index 1f1b441..0ecde0c 100644
--- a/Lib/ntpath.py
+++ b/Lib/ntpath.py
@@ -319,7 +319,7 @@ def expanduser(path):
     if 'USERPROFILE' in os.environ:
         userhome = os.environ['USERPROFILE']
     elif not 'HOMEPATH' in os.environ:
-        return path
+        return os.path.normpath(path)
     else:
         try:
             drive = os.environ['HOMEDRIVE']
@@ -346,7 +346,7 @@ def expanduser(path):
     if isinstance(path, bytes):
         userhome = os.fsencode(userhome)
 
-    return userhome + path[i:]
+    return os.path.normpath(userhome) + path[i:]
 
 
 # Expand paths containing shell variable substitutions.
-- 
2.35.1

